foreman/data_refinery_foreman/foreman/management/commands/organism_shepherd.py

```diff
@@ -0,0 +1,240 @@
+
+    prioritized_job_list = []
+    for experiment_stats_dict in completion_list:
+        unprocessed_samples = experiment_stats_dict["unprocessed"]
+
+        for sample in unprocessed_samples:
+            processor_jobs = list(sample.get_processor_jobs())
+            if processor_jobs:
+                # We want to requeue the most recently created
+                # processor job, so sort by id since they are
+                # autoincrementing.
+                processor_jobs.sort(
+                    reverse=True,
+                    key=lambda x: x.id
+                )
+                prioritized_job_list.append(processor_jobs[0])
+            else:
+                downloader_jobs = list(sample.get_downloader_jobs())
+                downloader_jobs.sort(
+                    reverse=True,
+                    key=lambda x: x.id
+                )
+                prioritized_job_list.append(downloader_jobs[0])
+
+    return prioritized_job_list
+
+
+def requeue_job(job):
+    """Requeues a job regardless of whether it is a DownloaderJob or ProcessorJob.
+
+    This function reuses a lot of logic from requeue_downloader_job
+    and requeue_processor_job from the main namespace for the
+    Foreman. However there's additional logic there that we don't
+    want, because we explicitly want to requeue these jobs regardless
+    of how many times they've been retried.
+    """
+    # All new jobs are going to be set at 2 retries so they only get
+    # tried once. Presumably all of these have failed at least once
+    # already, so there may be a good reason. Not immediately having
+    # them retried will give me a chance to actually take a look at
+    # what is happening to to them.
+    num_retries = 2
+    if type(job).__name__ == "ProcessorJob":
+        new_job = ProcessorJob(
+            num_retries=num_retries,
+            pipeline_applied=job.pipeline_applied,
+            ram_amount=job.ram_amount,
+            volume_index=job.volume_index
+        )
+        new_job.save()
+
+        for original_file in job.original_files.all():
+            ProcessorJobOriginalFileAssociation.objects.get_or_create(
+                processor_job=new_job,
+                original_file=original_file
+            )
+
+        job_type = ProcessorPipeline[job.pipeline_applied]
+    elif type(job).__name__ == "DownloaderJob":
+        new_job = DownloaderJob(
+            num_retries=num_retries,
+            downloader_task=job.downloader_task,
+            accession_code=job.accession_code
+        )
+        new_job.save()
+
+        for original_file in job.original_files.all():
+            DownloaderJobOriginalFileAssociation.objects.get_or_create(
+                downloader_job=new_job,
+                original_file=original_file
+            )
+
+        job_type = Downloaders[job.downloader_task]
+    else:
+        raise(ValueError("Told to requeue a job that's not a ProcessorJob nor DownloaderJob!"))
```

Comment:
Raise is a statement not a function, I'm also guessing this is an untested branch

Reply:
It is an untested branch because it _should never happen_. I don't actually know how I would get to this branch without changing the code to do so, it's just here as a just-in-case kinda thing.

Reply:
However I will use `raise` correctly.